### ISSUE TITLE
`Video(source="webcam")` doesn't work

### ISSUE THREAD

**Issue by pngwn** [gradio-app/gradio#2430]
### Describe the bug

As per title. Surprised this hasn't been reported already, unless this is not reproducible.

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Reproduction

```py
import gradio as gr

def video_identity(video):
    return video


demo = gr.Interface(
    video_identity,
    gr.Video(source="webcam"),
    "playable_video",
)

if __name__ == "__main__":
    demo.launch()
```

### Logs

```shell
/Users/peterallen/Projects/gradio/gradio/components.py:1689: UserWarning: Video does not have browser-compatible container or codec. Converting to mp4
  warnings.warn(
ffmpeg version 5.0.1 Copyright (c) 2000-2022 the FFmpeg developers
  built with Apple clang version 13.1.6 (clang-1316.0.21.2)
  configuration: --prefix=/opt/homebrew/Cellar/ffmpeg/5.0.1 --enable-shared --enable-pthreads --enable-version3 --cc=clang --host-cflags= --host-ldflags= --enable-ffplay --enable-gnutls --enable-gpl --enable-libaom --enable-libbluray --enable-libdav1d --enable-libmp3lame --enable-libopus --enable-librav1e --enable-librist --enable-librubberband --enable-libsnappy --enable-libsrt --enable-libtesseract --enable-libtheora --enable-libvidstab --enable-libvmaf --enable-libvorbis --enable-libvpx --enable-libwebp --enable-libx264 --enable-libx265 --enable-libxml2 --enable-libxvid --enable-lzma --enable-libfontconfig --enable-libfreetype --enable-frei0r --enable-libass --enable-libopencore-amrnb --enable-libopencore-amrwb --enable-libopenjpeg --enable-libspeex --enable-libsoxr --enable-libzmq --enable-libzimg --disable-libjack --disable-indev=jack --enable-videotoolbox --enable-neon
  libavutil      57. 17.100 / 57. 17.100
  libavcodec     59. 18.100 / 59. 18.100
  libavformat    59. 16.100 / 59. 16.100
  libavdevice    59.  4.100 / 59.  4.100
  libavfilter     8. 24.100 /  8. 24.100
  libswscale      6.  4.100 /  6.  4.100
  libswresample   4.  3.100 /  4.  3.100
  libpostproc    56.  3.100 / 56.  3.100
Input #0, mov,mp4,m4a,3gp,3g2,mj2, from '/var/folders/3w/6btg016509v7b2lz9h7vwqv00000gn/T/samplepnsnm0au.mp4':
  Metadata:
    major_brand     : isom
    minor_version   : 512
    compatible_brands: isomiso2avc1mp41
    encoder         : Lavf59.16.100
  Duration: 00:00:03.40, start: 0.000000, bitrate: 391 kb/s
  Stream #0:0[0x1](eng): Video: h264 (High) (avc1 / 0x31637661), yuv420p(progressive), 640x480 [SAR 1:1 DAR 4:3], 386 kb/s, 29.97 fps, 29.97 tbr, 30k tbn (default)
    Metadata:
      handler_name    : VideoHandler
      vendor_id       : [0][0][0][0]
File '/var/folders/3w/6btg016509v7b2lz9h7vwqv00000gn/T/samplepnsnm0au.mp4' already exists. Overwrite? [y/N] Not overwriting - exiting
Traceback (most recent call last):
  File "/Users/peterallen/Projects/gradio/gradio/routes.py", line 275, in run_predict
    output = await app.blocks.process_api(
  File "/Users/peterallen/Projects/gradio/gradio/blocks.py", line 791, in process_api
    predictions = self.postprocess_data(fn_index, result["prediction"], state)
  File "/Users/peterallen/Projects/gradio/gradio/blocks.py", line 759, in postprocess_data
    prediction_value = block.postprocess(prediction_value)
  File "/Users/peterallen/Projects/gradio/gradio/components.py", line 1696, in postprocess
    ff.run()
  File "/Users/peterallen/.pyenv/versions/3.10.1/lib/python3.10/site-packages/ffmpy.py", line 106, in run
    raise FFRuntimeError(self.cmd, self.process.returncode, out[0], out[1])
ffmpy.FFRuntimeError: `ffmpeg -i /var/folders/3w/6btg016509v7b2lz9h7vwqv00000gn/T/samplepnsnm0au.mp4 /var/folders/3w/6btg016509v7b2lz9h7vwqv00000gn/T/samplepnsnm0au.mp4` exited with status 1
```

### System Info

```shell
Latest `main`
```

### Severity

blocking all usage of gradio

**Comment by pngwn** [comment]
`"playable_video"` is not valid, maybe from before 3.0.

**Comment by freddyaboulton** [comment]
Is anyone else able to reproduce? The demo works for me 🤷‍♂️ 

![video_ffmpeg](https://user-images.githubusercontent.com/41651716/195160601-4f450e73-f8d0-4892-80a0-9fcd1f77c85f.gif)

**Comment by freddyaboulton** [comment]
I tried it with my macbook's built-in webcam and my cheap external webcam and both demos work for me.

**Comment by pngwn** [comment]
What version of ffmpeg?

**Comment by pngwn** [comment]
Also, a ghost seems to be using your computer. You might want to contact someone about that.

**Comment by freddyaboulton** [comment]
I'm on version 5.1.2! 

Thanks for the heads-up, I know just the right people to call

![](https://media.giphy.com/media/11jGtzDu7Nh89a/giphy.gif)

**Comment by abidlabs** [comment]
Works for me as well @pngwn with an ancient ffmpeg -- version 4.4!

**Comment by abidlabs** [comment]
Works for me as well @pngwn with an ancient ffmpeg -- version 4.4! (Running on Windows)

**Comment by pngwn** [comment]
Dunno whats going on with the webcam stuff, I'm getting wildly different behaviour to everyone else it seems.

**Comment by freddyaboulton** [comment]
@pngwn Does this still not work for you?

**Comment by abidlabs** [comment]
@pngwn let's close unless we can are able to repro?

**Comment by xingzhehe** [comment]
I got a similar error. I was uploading videos. In the first run, it works. In the second run, it fails.
I think the issue is in 
"""
File '/var/folders/3w/6btg016509v7b2lz9h7vwqv00000gn/T/samplepnsnm0au.mp4' already exists. Overwrite? [y/N] Not overwriting - exiting
"""
After I deleted the temporary mp4 file, it worked. But if I repeat uploading the same videos without deleting the existing file, the error shows again.
